src/charm.py

```diff
@@ -122,21 +125,38 @@ def is_main(self) -> bool:
         """
         return self.get_main_unit() == self.unit.name
 
-    def peer_addresses(self) -> list:
-        """Property to get addresses of the units in the cluster.
+    def instance_map(self) -> typing.Dict:
+        """Build instance_map config.
 
         Returns:
-            List of peer IP addresses.
+            Instance map configuration as a dict.
         """
-        addresses = []
+        unit_name = self.unit.name.replace("/", "-")
+        app_name = self.app.name
+        model_name = self.model.name
+        addresses = [f"{unit_name}.{app_name}-endpoints.{model_name}.svc.cluster.local"]
```

Comment:
Maybe just `addresses = [f"{unit_name}.{app_name}-endpoints"]`? (no need for the rest)